src/main/java/me/zeroeightsix/kami/util/astaralg/MyAstarAlg.java

```diff
@@ -0,0 +1,152 @@
+                        successor = new MyAstarNode(currentnode.pos.add(1, 0, 0), addlayer);
+                    }
+                    break;
+                case 1:
+                    if ((successor = grid.nodes.get(currentnode.pos.add(0, 0, 1))) == null) {
+                        successor = new MyAstarNode(currentnode.pos.add(0, 0, 1), addlayer);
+                    }
+                    break;
+                case 2:
+                    if ((successor = grid.nodes.get(currentnode.pos.add(-1, 0, 0))) == null) {
+                        successor = new MyAstarNode(currentnode.pos.add(-1, 0, 0), addlayer);
+
+                    }
+                    break;
+                case 3:
+                    if ((successor = grid.nodes.get(currentnode.pos.add(0, 0, -1))) == null) {
+                        successor = new MyAstarNode(currentnode.pos.add(0, 0, -1), addlayer);
+                    }
+                    break;
+                case 4:
+                    if ((successor = grid.nodes.get(currentnode.pos.add(1, 0, -1))) == null) {
+                        successor = new MyAstarNode(currentnode.pos.add(1, 0, -1), addlayer);
+                    }
+                    if (grid.nodes.get(currentnode.pos.add(1, 0, 0)).blocked || grid.nodes.get(currentnode.pos.add(0, 0, -1)).blocked)
+                        continue;
+                    break;
+                case 5:
+                    if ((successor = grid.nodes.get(currentnode.pos.add(-1, 0, -1))) == null) {
+                        successor = new MyAstarNode(currentnode.pos.add(-1, 0, -1), addlayer);
+                    }
+                    if (grid.nodes.get(currentnode.pos.add(-1, 0, 0)).blocked || grid.nodes.get(currentnode.pos.add(0, 0, -1)).blocked)
+                        continue;
+                    break;
+                case 6:
+                    if ((successor = grid.nodes.get(currentnode.pos.add(-1, 0, 1))) == null) {
+                        successor = new MyAstarNode(currentnode.pos.add(-1, 0, 1), addlayer);
+                    }
+                    if (grid.nodes.get(currentnode.pos.add(-1, 0, 0)).blocked || grid.nodes.get(currentnode.pos.add(0, 0, 1)).blocked)
+                        continue;
+                    break;
+                case 7:
+                    if ((successor = grid.nodes.get(currentnode.pos.add(1, 0, 1))) == null) {
+                        successor = new MyAstarNode(currentnode.pos.add(1, 0, 1), addlayer);
+                    }
+                    if (grid.nodes.get(currentnode.pos.add(1, 0, 0)).blocked || grid.nodes.get(currentnode.pos.add(0, 0, 1)).blocked)
+                        continue;
+                    break;
+
+            }
+
+            if (!closedlist.containsKey(successor.pos)) {
+                grid.nodes.put(successor.pos, successor);
+                if (!successor.blocked) {
+
+                    double tentative_g = currentnode.g + (i < 4 ? 1 : 1.41421);
```

Comment:
I'm assuming this number is just from wikipedia??

Reply:
this is the square root of 2  :D  when i>=4  it is a diagonal step hence we need to take the distance of traveling one block diagonal which is sqrt(1^2+1^2)=sqrt(2)~~1.41421